nixos/modules/services/misc/mpdscribble.nix

```diff
@@ -0,0 +1,292 @@
+# connect to.  Defaults to $MPD_PORT or 6600.
+port = ${cfg.port}
+
+${mkSection cfg "lastfm" "last.fm"}
+${mkSection cfg "librefm" "libre.fm"}
+${mkSection cfg "jamendo" "jamendo"}
+'';
+in
+
+{
+  ###### interface
+
+  options = {
+
+    services.mpdscribble = {
+
+      enable = mkOption {
+        default = false;
+        description = "
+          Whether to enable mpdscribble.
+        ";
+      };
+
+      proxy = mkOption {
+        default = null;
+        type = types.nullOr types.str;
+        description = ''
+          HTTP proxy URL.
+        '';
+      }
+
+      pidfile = mkOption {
+        default = "/var/run/mpdscribble.pid";
+        type = types.path;
+        description = ''
+          The location of the pid file. mpdscribble saves its process id there.
+        '';
+      }
+
+      daemonUser = mkOption {
+        default = "mpdscribble";
+        description = ''
+          Change to this system user after daemonization.
+        '';
+      }
+
+      logFile = mkOption {
+        default = "/dev/null";
+        example = "/tmp/mpdscribble.log";
+        description = ''
+          Location of the logfile.
+        '';
+      };
+
+      verbose = mkOption {
+        default = 1;
+        type = types.int;
+        description = ''
+          Log level for the mpdscribble daemon.
+        '';
+      };
+
+      journalInterval = mkOption {
+        default = 600;
+        example = 60;
+        type = types.int;
+        description = ''
+          How often should mpdscribble save the journal file? [seconds]
+        '';
+      };
+
+      host = mkOption {
+        default = "localhost";
+        type = types.str;
+        description = ''
+          Host for the mpdscribble daemon to search for a mpd daemon on.
+        '';
+      };
+
+      port = mkOption {
+        default = "localhost";
+        type = types.int;
+        description = ''
+          Port for the mpdscribble daemon to search for a mpd daemon on.
+        '';
+      };
+
+      lastfm = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = "
+            Whether to enable the mpdscribble daemon for lastfm.
+          ";
+        };
+
+        url  = mkOption {
+          default = false;
+          type = types.str;
+          description = "
+            Whether to enable the mpdscribble daemon for lastfm.
+          ";
+        };
+
+        username = mkOption {
+          default = "";
+          type = types.str;
+          description = "
+            Lastfm username.
+          ";
+        };
+
+        password = mkOption {
+          default = false;
+          type = types.str;
+          description = "
+            Lastfm Password. Attention: This will be stored in the globally
+            readable /nix/store.
+          ";
+        };
+
+        journal = mkOption {
+          default = "/tmp/lastfm-journal.mpdscribble";
```

Comment:
Is this journal persistent data? If yes, I would put into `/var` for example `/var/lib/mpdscribble`. If no, I would put it in `/run/` (if the amount of data stored is small).

Reply:
The amount is variable. mpdscribble stores scrobble data (about 5 lines of text, in human readable format) for each track played when it is not able to reach out to the last.fm servers... normally during network outage or when without internet connection. I'd opt for `/run` then.